tests/unit/legacy/api/test_json.py

```diff
@@ -118,6 +118,22 @@ def test_only_prereleases(self, monkeypatch, db_request):
         assert json_release.calls == [pretend.call(release, db_request)]
 
 
+class TestJSONProjectSlash:
+    def test_normalizing_redirects(self, db_request):
+        project = ProjectFactory.create()
+
+        name = project.name
+
+        db_request.matchdict = {"name": name}
+        db_request.route_path = pretend.call_recorder(
+            lambda *a, **kw: "/project/the-redirect"
+        )
+
+        resp = json.json_project_slash(project, db_request)
+
+        assert isinstance(resp, HTTPMovedPermanently)
```

Comment:
Can you add some more assertions here? Something like:

```python
assert db_request.route_path.calls == [
    call("legacy.api.json.project", name=project.name),
]
assert result.headers["Location"] == "/the-redirect"
```

Reply:
@stloma Can you add the assertion on `db_request.route_path.calls` as well?

Reply:
Ah, shoot, sorry I missed that :(